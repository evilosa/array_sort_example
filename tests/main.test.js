// 1) "103 123 4444 99 2000" => "2000 103 123 4444 99"
// 2) "2000 10003 1234000 44444444 9999 11 11 22 123" => "11 11 2000 10003 22 123 1234000 44444444 9999"

const testFunc = require('../src/index')

describe('Sort functions', () => {
  it('should throw the error when empty data send', () => {
    expect(() => { testFunc() }).toThrowErrorMatchingSnapshot()
  })

  it('should throw the TypeError when wrong data type sent', () => {
    expect(() => { testFunc(['123', '233']) }).toThrow(TypeError)
  })

  it('should provide correct results of work for case 1', () => {
    const source = '103 123 4444 99 2000'
    const result = '2000 103 123 4444 99'

    const funcResult = testFunc(source)
    expect(funcResult).toBe(result)
  })

  it('should provide correct results of work for case 2', () => {
    const source = '2000 10003 1234000 44444444 9999 11 11 22 123'
    const result = '11 11 2000 10003 22 123 1234000 44444444 9999'
 
    const funcResult = testFunc(source)
    expect(funcResult).toBe(result)
  })

  it('should provide correct results of work for case with broken numbers', () => {
    const source = '200$#as0 #$%@10003 12!3!f4qq000 4444rwr44asd44 9999 11 adf1d1fd 2asd2 1dsf23'
    const result = '11 11 2000 10003 22 123 1234000 44444444 9999'

    const funcResult = testFunc(source)
    expect(funcResult).toBe(result)
  })
})