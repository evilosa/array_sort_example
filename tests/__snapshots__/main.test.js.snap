// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort functions should throw the error when empty data send 1`] = `"Empty source string was provided!"`;
